web/docs/typescript.md

```diff
@@ -0,0 +1,264 @@
+---
```

Comment:
I experimented on how to best structure this, but putting it all in a single file turned out to make the most sense (at least to me).

Let me know if you have some ideas on how to improve it.